pkg/testframework/imageregistry.go

```diff
@@ -0,0 +1,220 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorapi "github.com/openshift/api/operator/v1alpha1"
+
+	imageregistryapi "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1alpha1"
+)
+
+const (
+	OperatorDeploymentNamespace      = "openshift-image-registry"
+	OperatorDeploymentName           = "cluster-image-registry-operator"
+	ImageRegistryName                = "image-registry"
+	ImageRegistryDeploymentName      = ImageRegistryName
+	ImageRegistryDeploymentNamespace = OperatorDeploymentNamespace
+)
+
+func ensureImageRegistryToBeRemoved() error {
+	cr := &imageregistryapi.ImageRegistry{
+		TypeMeta: metaapi.TypeMeta{
+			APIVersion: imageregistryapi.SchemeGroupVersion.String(),
+			Kind:       "ImageRegistry",
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name: ImageRegistryName,
+		},
+	}
+	if err := sdk.Get(cr); err != nil {
+		if errors.IsNotFound(err) {
+			// That's not exactly what we are asked for. And few seconds later
+			// the operator may bootstrap it. However, if the operator is
+			// disabled, it means the registry is not installed and we're
+			// already in the desired state.
+			return nil
+		}
+		return err
+	}
+
+	if err := sdk.Patch(cr, types.MergePatchType, []byte(`{"spec": {"managementState": "Removed"}}`)); err != nil {
+		return err
+	}
+
+	// TODO(dmage): when we have the Removed condition, this code will need to
+	// be updated.
+	time.Sleep(2 * time.Second)
+	return nil
+}
+
+func stopOperator() error {
+	deploy := &appsapi.Deployment{
+		TypeMeta: metaapi.TypeMeta{
+			APIVersion: appsapi.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name:      OperatorDeploymentName,
+			Namespace: OperatorDeploymentNamespace,
+		},
+	}
+	if err := sdk.Patch(deploy, types.MergePatchType, []byte(`{"spec": {"replicas": "0"}}`)); err != nil {
+		return err
+	}
+
+	return wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		deploy.Status.Replicas = 0
+		err = sdk.Get(deploy)
+		if err != nil {
+			return false, err
+		}
+		return deploy.Status.Replicas == 0, nil
+	})
+}
+
+func startOperator() error {
+	deploy := &appsapi.Deployment{
+		TypeMeta: metaapi.TypeMeta{
+			APIVersion: appsapi.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name:      OperatorDeploymentName,
+			Namespace: OperatorDeploymentNamespace,
+		},
+	}
+	if err := sdk.Patch(deploy, types.MergePatchType, []byte(`{"spec": {"replicas": "1"}}`)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteImageRegistryResource() error {
+	cr := &imageregistryapi.ImageRegistry{
+		TypeMeta: metaapi.TypeMeta{
+			APIVersion: imageregistryapi.SchemeGroupVersion.String(),
+			Kind:       "ImageRegistry",
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name: ImageRegistryName,
+		},
+	}
+
+	// TODO(dmage): the finalizer should be removed by the operator
```

Comment:
is this broken today for some reason?

Reply:
I don't think there's any reason. Maybe @legionus knows.